src/main/webapp/app/exercises/shared/exercise-headers/exercise-headers-information/exercise-headers-information.component.ts

```diff
@@ -0,0 +1,286 @@
+            this.informationBoxItems.push(startDateItem);
+        }
+    }
+
+    addDifficultyItem() {
+        if (this.exercise.difficulty) {
+            const difficultyItem: InformationBox = {
+                title: 'artemisApp.courseOverview.exerciseDetails.difficulty',
+                content: {
+                    type: 'difficultyLevel',
+                    value: this.exercise.difficulty,
+                },
+                isContentComponent: true,
+            };
+            this.informationBoxItems.push(difficultyItem);
+        }
+    }
+
+    addSubmissionStatusItem() {
+        const submissionStatusItem: InformationBox = {
+            title: 'artemisApp.courseOverview.exerciseDetails.status',
+            content: {
+                type: 'submissionStatus',
+                value: this.exercise,
+            },
+            isContentComponent: true,
+        };
+        this.informationBoxItems.push(submissionStatusItem);
+    }
+
+    addCategoryItems() {
+        const notReleased = this.exercise.releaseDate && dayjs(this.exercise.releaseDate).isAfter(dayjs());
+
+        if (notReleased || this.exercise.includedInOverallScore !== IncludedInOverallScore.INCLUDED_COMPLETELY || this.exercise.categories?.length) {
+            const categoryItem: InformationBox = {
+                title: 'artemisApp.courseOverview.exerciseDetails.categories',
+                content: {
+                    type: 'categories',
+                    value: this.exercise,
+                },
+                isContentComponent: true,
+            };
+            this.informationBoxItems.push(categoryItem);
+        }
+    }
+
+    addSubmissionPolicyItem() {
+        if (this.submissionPolicy?.active && this.submissionPolicy?.submissionLimit) {
+            this.informationBoxItems.push(this.getSubmissionPolicyItem());
+        }
+    }
+
+    getSubmissionPolicyItem(): InformationBox {
+        return {
+            title: 'artemisApp.programmingExercise.submissionPolicy.submissionLimitTitle',
+            content: {
+                type: 'string',
+                value: `${this.numberOfSubmissions} /  ${this.submissionPolicy?.submissionLimit}`,
+            },
+            contentColor: this.submissionPolicy?.submissionLimit ? this.getSubmissionColor() : 'body-color',
+            tooltip: 'artemisApp.programmingExercise.submissionPolicy.submissionPolicyType.' + this.submissionPolicy?.type + '.tooltip',
+            tooltipParams: { points: this.submissionPolicy?.exceedingPenalty?.toString() },
+        };
+    }
+
+    getSubmissionColor(): string {
+        const submissionsLeft = this.submissionPolicy?.submissionLimit ? this.submissionPolicy?.submissionLimit - this.numberOfSubmissions : 2;
+        if (submissionsLeft > 1) return 'body-color';
+        else {
+            return submissionsLeft <= 0 ? 'danger' : 'warning';
+        }
+    }
+
+    getPointsItem(points: number | undefined, title: string): InformationBox {
```

Comment:
Why can `points` be undefined here and how does the method handle this case then?